cdap-app-fabric/pom.xml

```diff
@@ -264,6 +264,54 @@
       <groupId>org.slf4j</groupId>
       <artifactId>log4j-over-slf4j</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-source-control</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-source-control</artifactId>
+      <version>${project.version}</version>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.eclipse.jgit</groupId>
+      <artifactId>org.eclipse.jgit.http.server</artifactId>
+      <version>${jgit.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-security</artifactId>
```

Comment:
why are all these jetty dependencies needed?

Reply:
These dependencies are required in the git relates unit tests, like we need them to run a local git server

Reply:
Specifically they're needed in `LocalGitServer`: https://github.com/cdapio/cdap/blob/8c0e7f50df0cc0f63e9e6ed0e15e6b5fe0e8719d/cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/LocalGitServer.java, which is used in the `SourceControlManagementHttpHandlerTests`

Reply:
The HTTP handler test doesn't need a real git server for testing, right? Can we mock/stub the interface instead?

Reply:
Or can we have the SourceControlManagementHttpHandler sits inside the source control maven module?

Reply:
@chtyim
1. For the question if `SourceControlManagementHttpHandler` can sit in scm module, it brings circular dependency issue because it needs to interact with `ApplicationLifecycleService` from appFabric for application push and pull/deploy
2. Let me try mocking the `RepositoryManager.validateConfig` method instead

Reply:
So currently most of the handlers tests in  AppFabric extend `AppFabricTestBase`, which is basically running appFabric servers instead of using  mock/stubs... Is there a way that we could inject the mocked instance into the actual running AppFabricTestBase? OR should we drop the convention and write the tests in the mocked fashion(provided mocked instance in constructors, call and expect the actual handler methods etc)?

Reply:
you can setup the guice module in the test to bind SourceControlManagementService and SecureStore to some mock/fake implementation.

In addition, testing business logic should mostly be done using tests for the underlying service. The HTTP tests should target logic specific to the handler, like what happens with bad input, etc.

Reply:
@chtyim @albertshau Thanks for the idea! I've made a Mockito.spy on SourceControlManagementService in the tests and reverted the unnecessary changes. Please have another round of review